traffic_ops/testing/api/v4/profiles_export_test.go

```diff
@@ -0,0 +1,90 @@
+	"github.com/apache/trafficcontrol/traffic_ops/toclientlib"
+)
+
+func TestProfilesExport(t *testing.T) {
+	WithObjs(t, []TCObj{CDNs, Types, Parameters, Profiles, ProfileParameters}, func() {
+
+		methodTests := utils.V4TestCase{
+			"GET": {
+				"OK when VALID request": {
+					EndpointId:    GetProfileID(t, "EDGE1"),
+					ClientSession: TOSession,
+					Expectations: utils.CkRequest(utils.NoError(), utils.HasStatus(http.StatusOK),
+						validateProfilesExportFields(map[string]interface{}{"ProfileCDNName": "cdn1", "ProfileName": "EDGE1",
+							"ProfileDescription": "edge1 description", "ProfileType": "ATS_PROFILE"})),
+				},
+				"NOT FOUND when PROFILE DOESNT EXIST": {
+					EndpointId:    func() int { return 1111111111 },
+					ClientSession: TOSession,
+					Expectations:  utils.CkRequest(utils.HasError(), utils.HasStatus(http.StatusNotFound)),
+				},
+			},
+		}
+
+		for method, testCases := range methodTests {
+			t.Run(method, func(t *testing.T) {
+				for name, testCase := range testCases {
+					switch method {
+					case "GET":
+						t.Run(name, func(t *testing.T) {
+							resp, reqInf, err := testCase.ClientSession.ExportProfile(testCase.EndpointId(), testCase.RequestOpts)
+							for _, check := range testCase.Expectations {
+								check(t, reqInf, resp, resp.Alerts, err)
+							}
+						})
+					}
+				}
+			})
+		}
+
+	})
+}
+
+func validateProfilesExportFields(expectedResp map[string]interface{}) utils.CkReqFunc {
+	return func(t *testing.T, _ toclientlib.ReqInf, resp interface{}, _ tc.Alerts, _ error) {
+		assert.RequireNotNil(t, resp, "Expected Profiles Export response to not be nil.")
+		profileExport := resp.(tc.ProfileExportResponse)
+		for field, expected := range expectedResp {
+			switch field {
+			case "ProfileCDNName":
+				assert.RequireNotNil(t, profileExport.Profile.CDNName, "Expected Profile CDNName to not be nil.")
+				assert.Equal(t, expected, *profileExport.Profile.CDNName, "Expected Profile.CDNName to be %v, but got %d", expected, *profileExport.Profile.CDNName)
+			case "ProfileDescription":
+				assert.RequireNotNil(t, profileExport.Profile.Description, "Expected Profile Description to not be nil.")
+				assert.Equal(t, expected, *profileExport.Profile.Description, "Expected Profile.Description to be %v, but got %d", expected, *profileExport.Profile.Description)
+			case "ProfileName":
+				assert.RequireNotNil(t, profileExport.Profile.Name, "Expected Profile Name to not be nil.")
+				assert.Equal(t, expected, *profileExport.Profile.Name, "Expected Profile.Name to be %v, but got %d", expected, *profileExport.Profile.Name)
+			case "ProfileType":
+				assert.RequireNotNil(t, profileExport.Profile.Type, "Expected Profile Type to not be nil.")
+				assert.Equal(t, expected, *profileExport.Profile.Type, "Expected Profile.Type to be %v, but got %d", expected, *profileExport.Profile.Type)
+			default:
+				t.Errorf("Expected field: %v, does not exist in response", field)
+			}
```

Comment:
These fields are all strings, and you already have the field name. So instead of a switch statement, you could use

```go
fieldValue := reflect.Indirect(reflect.ValueOf(profileExport.Profile).FieldByName(field)).String()
```
to get the field value and only write `assert.RequireNotNil() + assert.Equal()` once.

Note that the field name strings would need to be changed for this to work (`"ProfileCDNName"` -> `"CDNName"`, `"ProfileDescription"` -> `"Description"`, etc.)

Reply:
The cost of this change would be readability and isn't reflection expensive? I am not sure its worth it just to reduce the amount of lines of code.

Reply:
> The cost of this change would be readability
Agreed that the `reflect` line is non-obvious at first glance, maybe add a comment to explain it, or break it down into more than 1 line?

> isn't reflection expensive

Not to the point that it just shouldn't be used. The time complexity of a case statement approaches O(n), while the time complexity of reflection is O(1)

Besides, the slowest part of the API tests is the TO API itself (which also uses reflection), not the tests